internal/controller/apiextensions/composite/reconciler.go

```diff
@@ -608,6 +645,22 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		if kerrors.IsConflict(err) {
 			return reconcile.Result{Requeue: true}, nil
 		}
+
+		meta := r.handleCommonCompositionResult(ctx, res, xr, log)
+		// We encountered a fatal error. For any custom status conditions that were
+		// not received due to the fatal error, mark them as unknown.
+		for _, c := range xr.GetConditions() {
+			if xpv1.IsSystemConditionType(c.Type) {
+				continue
+			}
+			if _, ok := meta.conditionTypesSeen[c.Type]; !ok {
+				c.Status = corev1.ConditionUnknown
+				c.Reason = reasonFatalError
+				c.Message = "Crossplane could not reach the code that updates this condition because it hit a fatal error."
```

Comment:
```suggestion
				c.Message = "Crossplane didn't call the function that updates this condition because an earlier function in the pipeline returned a fatal error."
```

Playing around with wording to try make this as clear as possible. Let me know what you think.

Reply:
This could be misleading if the function itself returns with a fatal error before setting the condition.

What about: "A fatal error occurred before the status of this condition could be determined.".

Reply:
@dalton-hill-0 True, good point.

I like your suggestion. It would be ideal to provide breadcrumbs to where the fatal error was though - i.e. in a function. Perhaps "A function returned a fatal error before..."?

Reply:
I agree it would be ideal to include where it came from, but we cannot guarantee that without parsing the error in this case. If we parse to check for `errFmtFatalResult`, we could have one message/reason for failed function results, and another for crossplane system errors. Thoughts?

Reply:
@dalton-hill-0 Ah, I was thinking this could only happen if a function returned a fatal result. I suppose it could happen for other errors returned from the FunctionComposer too. A little part of me still feels like it's kinda overkill to update these status conditions at all in the face of a fatal error, but I know we've discussed that a lot.

"A fatal error occurred before the status of this condition could be determined" sounds good to me.